src/coreneuron/utils/randoms/nrnran123.cpp

```diff
@@ -146,15 +150,15 @@ void nrnran123_set_globalindex(uint32_t gix) {
 void nrnran123_initialise_global_state_on_device() {
     if (coreneuron::gpu_enabled()) {
 #ifndef __CUDACC__
```

Comment:
I haven't understood with what cmake options is this macro enabled to try my changes.
Any help @olupton ?

Reply:
I believe `__CUDACC__` **is** defined by `nvc++ -cuda` (so this branch is **not** taken).
I don't think we're actively testing a toolchain where this does something meaningful (i.e. GPU offload without `nvc++`).